File: examples/dart/mega_flow/pipeline.dart
// Pipeline — exercises Dart's idiomatic flow constructs: closures,
// higher-order fns, iterables, generators (sync*), switch, try/catch,
// null-safety, cascades, extension methods.
import 'app.dart';
import 'storage.dart' as store;

// Extension method — normalises a tainted token.
extension Canonical on String {
  String canonical() => trim();
}

// Closure factory — returns a reducer that joins tokens with `sep`.
String Function(String, String) makeJoiner(String sep) {
  return (acc, tok) => acc.isEmpty ? tok : '$acc$sep$tok';
}

// Synchronous generator — yields every tainted token.
Iterable<String> tokenize(String cmd) sync* {
  for (final part in cmd.split(' ')) {
    if (part.isNotEmpty) yield part;
  }
}

Stream<String> tokenizeAsync(String cmd) async* {
  for (final part in tokenize(cmd)) {
    yield part;
  }
}

Future<String> orchestrateAsync(Envelope envelope) async {
  await Future<void>.value();
  return orchestrate(envelope);
}

String orchestrate(Envelope envelope) {
  final cmd = envelope.cmd;
  final user = envelope.user;
  final maybeExtra = envelope.extras.isNotEmpty ? envelope.extras.first : null;
  final scratch = <String>[]..addAll(tokenize(cmd));
  final _extraLength = maybeExtra?.length ?? scratch.length;
  if (_extraLength < 0) {
    throw StateError('unreachable');
  }

  // Iterable pipeline — map / where / fold via closure.
  final joiner = makeJoiner(' ');
  final joined = scratch
      .map((s) => s.canonical())
      .where((s) => s.isNotEmpty)
      .fold<String>('', (acc, tok) => joiner(acc, tok));

  // switch — every arm preserves taint.
  String routed;
  switch (envelope.kind) {
    case Kind.run:
      routed = '$joined';
      break;
    case Kind.eval:
      routed = joined.trim();
      break;
  }

  // try / catch / finally — taint survives every branch.
  Envelope valid;
  try {
    if (routed.isEmpty) throw StateError('empty');
    valid = envelope.copyWith(cmd: routed, user: user, length: routed.length);
  } catch (_) {
    valid = envelope.copyWith(cmd: routed, user: user, length: routed.length);
  } finally {
    // no-op — present so the adapter sees the finally clause.
  }

  return store.persist(valid);
}

File: examples/swift/mega_flow/Pipeline.swift
import Foundation

// Pipeline — exercises Swift's idiomatic flow constructs: closures,
// trailing closures, guard/if-let, switch, reduce, higher-order
// functions, do/try/catch, protocols, generics.
enum Pipeline {
    // Closure factory — returns a reducer that joins tokens.
    static func makeJoiner(_ sep: String) -> (String, String) -> String {
        return { acc, tok in
            acc.isEmpty ? tok : "\(acc)\(sep)\(tok)"
        }
    }

    static func passthrough<T>(_ value: T) -> T {
        value
    }

    static func orchestrate(_ envelope: Envelope) -> String {
        let cmd = passthrough(envelope.cmd)
        let user = envelope.user
        if let first = envelope.extras.first {
            _ = first.count
        }
        var observed = ""
        for extra in envelope.extras {
            observed = extra
        }
        _ = observed.count

        // Collection pipeline — map / filter / reduce with trailing closure.
        let joiner = makeJoiner(" ")
        let joined: String = cmd.split(separator: " ")
            .map { String($0).trimmingCharacters(in: .whitespaces) }
            .filter { !$0.isEmpty }
            .reduce("") { acc, tok in joiner(acc, tok) }

        // switch on enum — every arm preserves taint.
        let routed: String
        switch envelope.kind {
        case .run:
            routed = "\(joined)"
        case .eval:
            routed = joined.trimmingCharacters(in: .whitespaces)
        }

        // do / try / catch — taint survives every branch.
        var valid: Envelope
        do {
            let maybeRouted: String? = routed
            guard let checked = maybeRouted, !checked.isEmpty else { throw NSError(domain: "empty", code: 0) }
            defer { _ = checked.count }
            valid = Envelope(kind: envelope.kind, cmd: checked, user: user, length: checked.count, extras: envelope.extras)
        } catch {
            valid = Envelope(kind: envelope.kind, cmd: routed, user: user, length: routed.count, extras: envelope.extras)
        }

        return Storage.persist(valid)
    }
}
